tests/unit/test_charm.py

```diff
@@ -131,3 +132,84 @@ def test__on_get_admin_password_action(
     jenkins_charm._on_get_admin_password(mock_event)
 
     mock_event.set_results.assert_called_once_with({"password": admin_credentials.password})
+
+
+def test__on_update_status_no_action(
+    harness_container: HarnessWithContainer, monkeypatch: pytest.MonkeyPatch, current_version: str
+):
+    """
+    arrange: given latest jenkins, monkeypatched jenkins that returns the latest patch version.
+    act: when update_status action is triggered.
+    assert: no action is taken and the charm remains active.
+    """
+    mock_download_func = MagicMock(spec=jenkins.download_stable_war)
+    monkeypatch.setattr(jenkins, "get_version", lambda: current_version)
+    monkeypatch.setattr(jenkins, "get_latest_patch_version", lambda *_, **__: current_version)
+    monkeypatch.setattr(jenkins, "download_stable_war", mock_download_func)
+    mock_event = MagicMock(spec=UpdateStatusEvent)
+    harness_container.harness.begin()
+
+    jenkins_charm = cast(JenkinsK8SOperatorCharm, harness_container.harness.charm)
+    jenkins_charm._on_update_status(mock_event)
+
+    mock_download_func.assert_not_called()
+    assert jenkins_charm.unit.status.name == ACTIVE_STATUS_NAME
+
+
+def test__on_update_status_error(
+    harness_container: HarnessWithContainer,
+    monkeypatch: pytest.MonkeyPatch,
+    versions: Versions,
+    raise_exception: Callable,
+    caplog: pytest.LogCaptureFixture,
+):
+    """
+    arrange: given unpatched jenkins, monkeypatched download_stable_war that raises an exception.
+    act: when update_status action is triggered.
+    assert: no action is taken and the charm remains active but the error is logged.
+    """
+    monkeypatch.setattr(jenkins, "get_version", lambda: versions.current)
+    monkeypatch.setattr(jenkins, "get_latest_patch_version", lambda *_, **__: versions.patched)
+    monkeypatch.setattr(
+        jenkins,
+        "download_stable_war",
+        lambda *_args, **_kwargs: raise_exception(exception=jenkins.JenkinsNetworkError),
```

Comment:
`download_stable_war` could also raise a `ValidationError` which isn't covered, and probably work as expected

Reply:
Added test cases for those! Thanks for catching that!